extract svg package

oliverpool opened the pull request:
I added one method (following our discussion in #36):
```go
func (t *Text) WalkSpans(cb func(y, dx float64, span TextSpan)) {
	for _, line := range t.lines {
		for _, span := range line.spans {
			cb(line.y, span.dx, span)
		}
	}
}
```

I previously overlooked a bunch of stuff that need to be exported from the FontFace as well
- color
- scale
- voffset

Actually `FontFace.Info()` does not seem to be used anywhere, I could:
- either expand it to include the 3 properties above
- remove it and export the need properties (and adding a `Name()` property to prevent having to export `Family`)

What do you think?

tdewolff commented on text.go:
```diff
@@ -542,7 +543,7 @@ func (t *Text) ToPaths() ([]*Path, []color.RGBA) {
 			p := deco.ff.Decorate(deco.x1 - deco.x0)
 			p = p.Translate(deco.x0, line.y)
 			paths = append(paths, p)
-			colors = append(colors, deco.ff.color)
```
For consistency we should probably rename this to `deco.face.Color` or `deco.Face.Color`.

oliverpool replied:
renamed!

tdewolff commented on svg/util.go:
```diff
@@ -0,0 +1,42 @@
+)
+
+// equal returns true if a and b are equal with tolerance Epsilon.
+func equal(a, b float64) bool {
```
We could probably export this from `canvas`  as `Equal`.

oliverpool replied:
exported!

tdewolff commented:
Removing `.Info()` if probably a good idea, doesn't really serve much.

oliverpool commented:
I removed the `Info` (and added a `Name()` method), as well as implemented your other suggestions.

I restricted myself to `svg` in this PR, since it already implies a lot of changes.

You are right that `pdf` will require a solution to access the boundaries, but I prefer to tackle this in an other PR.
- Either you think that this one is good enought to be merged, and I will make a fresh PR for `pdf`
- Or you prefer to merge both at the same time in master, and I will do a PR to merge on this branch (so that it is easier to review the changes).

tdewolff commented:
Nice! Let's merge this one now, can you fix the failing check so I can merge?

oliverpool commented:
> can you fix the failing check so I can merge?

[Travis-ci](https://travis-ci.org/github/tdewolff/canvas/pull_requests) looks green, only the coverage is down by 0.3% (because the amount of code slightly increased). Where should I add some coverage ?

tdewolff commented:
Ah not sure why I couldn't merge yesterday. Coverage is not so important, we'll improve that later.
